imports/plugins/core/hydra-oauth/server/util/hydra.js

```diff
@@ -78,12 +78,36 @@ function deleteUserSession(id) {
     });
 }
 
+/**
+ * @name refreshAuthToken
+ * @method
+ * @private
+ * @param  {String} options options
+ * @return {Object|String} API res
+ */
+function refreshAuthToken({ refreshToken, clientId, clientSecret }) {
+  return fetch(`${HYDRA_TOKEN_URL}`, {
+    headers: { "Content-Type": "application/x-www-form-urlencoded" },
+    method: "POST",
+    body: `grant_type=refresh_token&refresh_token=${refreshToken}&response_type=token&client_id=${clientId}&client_secret=${clientSecret}`
+  })
+    .then(async (res) => {
+      if (res.status < 200 || res.status > 302) {
+        const json = await res.json();
+        Logger.error("An error occurred while calling refresh API", json.error_description);
+        return Promise.reject(new Error(json.error_description));
+      }
+      return res.json();
+    });
+}
+
 export default {
   getLoginRequest: (challenge) => get("login", challenge),
   acceptLoginRequest: (challenge, body) => put("login", "accept", challenge, body),
   rejectLoginRequest: (challenge) => put("login", "reject", challenge),
   getConsentRequest: (challenge) => get("consent", challenge),
   acceptConsentRequest: (challenge, body) => put("consent", "accept", challenge, body),
   rejectConsentRequest: (challenge, body) => put("consent", "reject", challenge, body),
-  deleteUserSession: (id) => deleteUserSession(id)
+  deleteUserSession: (id) => deleteUserSession(id),
+  refreshAuthToken: (options) => refreshAuthToken(options)
```

Comment:
For these last two, they are not adding any arguments and are named the same, so you can eliminate a function layer by instead doing:

```js
// ...
  deleteUserSession,
  refreshAuthToken
}
```